File: Task-03/prime.ex
defmodule Prime do
  def is_prime(num) when num < 2, do: false
  def is_prime(num) when num == 2, do: true
  def is_prime(num) do
    Enum.all?(2..trunc(:math.sqrt(num)), fn i -> rem(num, i) != 0 end)
  end
end

IO.puts("Enter a number: ")
n = String.to_integer(IO.gets(""))

primes = Enum.filter(2..n, &Prime.is_prime/1)
IO.puts("Prime numbers up to #{n} are: #{inspect primes}")

File: Task-03/prime.go
package main

import (
	"fmt"
	"math"
)

func isPrime(num int) bool {
	if num < 2 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(num))); i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var n int
	fmt.Print("Enter a number: ")
	fmt.Scan(&n)

	primes := []int{}
	for i := 2; i <= n; i++ {
		if isPrime(i) {
			primes = append(primes, i)
		}
	}

	fmt.Printf("Prime numbers up to %d are: %v\n", n, primes)
}

